app/Http/Controllers/Auth/VerifyMobileController.php

```diff
@@ -17,24 +18,42 @@ public function __invoke(Request $request)
             'code' => ['required', 'numeric'],
         ]);
 
+        // If the minutes_of_validation feature is enabled 
+        if(config('mobile.minutes_of_validation') > 0) {
+            $lastSend = Carbon::parse($request->user()->mobile_last_send);
+            if($lastSend->diffInMinutes(Carbon::now()) > config('mobile.minutes_of_validation')) {
+                $request->user()->sendMobileVerificationNotification(true);
+                return back()->with(['error' => __('mobile.expired')]);
+            }
+        }
+
+        // Code correct
         if ($request->code === auth()->user()->mobile_verify_code) {
             $request->user()->markMobileAsVerified();
-
             return redirect()->to(RouteServiceProvider::HOME)->with(['message' => __('mobile.verified')]);
         }
 
-        if($request->user()->mobile_attempts_left <= 0) {
-            $request->user()->update([
-                'mobile_verify_code' => random_int(111111, 999999),
-                'mobile_attempts_left' => config('mobile.max_attempts'),
-            ]);
-            $request->user()->sendMobileVerificationNotification();
-            return back()->with(['error' => __('mobile.new_code')]);
-        }
-        else {
+        // Max attempts feature
+        if(config('mobile.max_attempts') > 0) {
```

Comment:
@MaxRomagnoli can you check code styling with PSR standards? Pretty sure it should be space after if: `if (config...)`